src/Settings.js

```diff
@@ -0,0 +1,172 @@
+import { validate } from '../packages/device-api'
+import {GetAvailableInputTypesCall, GetRuntimeConfigurationCall} from './deviceApiCalls/__generated__/deviceApiCalls'
+import {autofillSettingsSchema} from './deviceApiCalls/__generated__/validators.zod'
+
+/**
+ * Some Type helpers to prevent duplication
+ * @typedef {import("./deviceApiCalls/__generated__/validators-ts").AutofillFeatureToggles} AutofillFeatureToggles
+ * @typedef {import("./deviceApiCalls/__generated__/validators-ts").AvailableInputTypes} AvailableInputTypes
+ * @typedef {import("../packages/device-api").DeviceApi} DeviceApi
+ */
+
+/**
+ * The Settings class encapsulates the concept of 1) feature toggles + 2) available input types.
+ *
+ * 1) Feature toggles are boolean flags that can represent a device's capabilities. That may be user-toggled
+ * or not, we don't make that distinction.
+ *
+ * 2) Available Input Types are indicators to whether the given platform can provide data for a given type.
+ * For example, a user might have credentials saved for https://example.com, so when the page loads, but **before**
+ * we can decorate any fields, we determine this first.
+ */
+export class Settings {
+    /** @type {GlobalConfig} */
+    globalConfig;
+    /** @type {DeviceApi} */
+    deviceApi;
+    /** @type {AutofillFeatureToggles | null} */
+    _featureToggles = null
+    /** @type {AvailableInputTypes | null} */
+    _availableInputTypes = null
+
+    /**
+     * @param {GlobalConfig} config
+     * @param {DeviceApi} deviceApi
+     */
+    constructor (config, deviceApi) {
+        this.deviceApi = deviceApi
+        this.globalConfig = config
+    }
+
+    /**
+     * Feature toggles are delivered as part of the Runtime Configuration - a flexible design that
+     * allows data per user + remote config to be accessed together.
+     *
+     * Once we access the Runtime Configuration, we then extract the autofill-specific settings via
+     * `runtimeConfig.userPreferences.features.autofill.settings` and validate that separately.
+     *
+     * The 2-step validation occurs because RuntimeConfiguration will be coming from a shared library
+     * and does not know about the shape of Autofill specific settings.
+     *
+     * @returns {Promise<AutofillFeatureToggles>}
+     */
+    async getFeatureToggles () {
+        try {
+            const runtimeConfig = await this.deviceApi.request(new GetRuntimeConfigurationCall(null))
+            const autofillSettings = validate(runtimeConfig.userPreferences?.features?.autofill?.settings, autofillSettingsSchema)
+            return autofillSettings.featureToggles
+        } catch (e) {
+            // these are the fallbacks for when a platform hasn't implemented the calls above. (like on android)
+            if (this.globalConfig.isDDGTestMode) {
```

Comment:
@GioSensation yep, an injected logger to the device is something we should do - let's make a follow up task as I didn't want to introduce any more new concepts here :)

In terms of logging in the catch, actually this is the correct flow here - the calls *will* be made on all platforms, but the transport will throw if it can't fulfil it (either by talking to the device or by reading config).

For example, see: 

https://github.com/duckduckgo/duckduckgo-autofill/blob/14c9aa25d08658ff1f83ffa4cf8eaed66ee47be3/src/deviceApiCalls/transports/android.transport.js#L91-L93

This means that the log message you mentioned above *would* give good context about the error, but only in development/test mode